From 27abd90f714fe760b10cb3724e6a515ccfebbd01 Mon Sep 17 00:00:00 2001
From: Lijun Fang <fanglijun3@huawei.com>
Date: Tue, 22 Jun 2021 08:04:42 +0000
Subject: [PATCH] iommu: sva: Fix compile error in iommu_sva_bind_group
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 27abd90f714fe760b10cb3724e6a515ccfebbd01


hulk inclusion
category: bugfix
bugzilla: 51855
CVE: NA

---------------------------------------------

iommu_sva_bind_group should return pointer rather than an integer,
so it will cause compile error, if not defined CONFIG_IOMMU_API

Fixes: 45e52e3fc545 ("iommu: Add group variant for SVA bind/unbind")
Signed-off-by: Lijun Fang <fanglijun3@huawei.com>
Reviewed-by: Weilong Chen <chenweilong@huawei.com>
Signed-off-by: Chen Jun <chenjun102@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/iommu.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index 30795e2b15a5..3ff424d4f481 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -1114,7 +1114,7 @@ static inline struct iommu_sva *
 iommu_sva_bind_group(struct iommu_group *group, struct mm_struct *mm,
 		     void *drvdata)
 {
-	return -ENODEV;
+	return NULL;
 }
 
 #endif /* CONFIG_IOMMU_API */
-- 
2.26.2

